python-lib/ontology_tagging_formatting.py

```diff
@@ -38,18 +38,32 @@ def _arrange_columns_order(
             df.reset_index(inplace=True)
         return df
 
+    def _apply_matcher(
+        self, row: pd.Series, language_column: AnyStr
+    ) -> Tuple[AnyStr, List]:
+        """Apply matcher to document in the given row and returns it with the associated language"""
+        language = (
+            row[language_column]
+            if self.language == "language_column"
+            else self.language
+        )
+        document = list(
+            self.nlp_dict[language].pipe(row[self.splitted_sentences_column])
+        ) 
+        return language, document
+
 
 class FormatterByTag(Formatter):
     def __init__(self, *args, **kwargs):
         super(FormatterByTag, self).__init__(*args, **kwargs)
 
-    def write_df_category(self, input_df, text_column) -> pd.DataFrame:
-        return self.write_df(input_df, text_column)
+    def write_df_category(self, input_df, text_column, language_column) -> pd.DataFrame:
```

Comment:
What about the case where there is a single language, but no language column?

Reply:
language_column would be None, I clarified that in ux/imrovments PR

Reply:
To clarify even further, you can use `from typing import Optional` (https://stackoverflow.com/questions/51710037/how-should-i-use-the-optional-type-hint)

Also, for next time, I prefer to avoid mixing commits and comments between PRs.